### PR TITLE
Added so library test target for MQTT

### PR DESCRIPTION
**Pull request by bernhard-thiele**
The goal is to ensure that the shipped MQTT libraries can be used for building
shared libraries which might be needed for building FMUs, see #306.

### CODE REVIEW

File: Modelica_DeviceDrivers/Resources/test/Communication/test_MDDMQTT_buildAsSharedLibrary.c [hunk 1]
```diff
@@ -0,0 +1 @@
+#include "../../Include/MDDMQTT.h"
```
Comment (beutlich) [inline Modelica_DeviceDrivers/Resources/test/Communication/test_MDDMQTT_buildAsSharedLibrary.c]: One more thought: Is this actually enough? Or should there be some implementation?

Comment (bernhard-thiele) [reply Modelica_DeviceDrivers/Resources/test/Communication/test_MDDMQTT_buildAsSharedLibrary.c]: Think it should be enough to catch the error. I tested once with the old libraries and got the expected error, when I tested with your new commit and it worked.

Comment (beutlich) [reply Modelica_DeviceDrivers/Resources/test/Communication/test_MDDMQTT_buildAsSharedLibrary.c]: OK, that's the information I was missing. Thanks.

Comment (bernhard-thiele) [reply Modelica_DeviceDrivers/Resources/test/Communication/test_MDDMQTT_buildAsSharedLibrary.c]: It won't give an error during running the tests, it either fails or passes during compilation. However, a build fail is detected by travis and so we get the information.

**Pull request merged**